server/src/controllers/authentication.controller.ts

```diff
@@ -26,8 +28,66 @@ class AuthenticationController implements Controller {
     this.router.post(`${this.path}/register`, validationMiddleware(CreateUserDto), this.registration);
     this.router.post(`${this.path}/login`, validationMiddleware(LogInDto), this.loggingIn);
     this.router.post(`${this.path}/logout`, this.loggingOut);
+    this.router.get(`${this.path}/confirm/:confirmationCode`, this.verifyUser);
+    this.router.post(`${this.path}/forgot`, this.forgotPassword);
+    this.router.patch(`${this.path}/forgetConfirm`, this.newPassword)
   }
 
+  private newPassword = async(request: Request, response: Response, next: NextFunction) => {
+    const thisRequest = this.user.findOne({
+      resetPasswordToken: request.body.id
+    })
+
+    if(thisRequest){
+      const newPassword = request.body.password;
+      const hashedPassword = await bcrypt.hash(newPassword, +process.env.SALT);
+
+      this.user.findOne({
+        resetPasswordToken: request.body.id,
+      })
+        .then((user) => {
+          if (!user) {
+            return response.status(404).send({ message: "User Not found." });
+          }
+    
+          user.password = hashedPassword;
+          user.save((err) => {
+            if (err) {
+              response.status(500).send({ message: err });
+              return;
+            }
+          });
+        })
+        .catch((e) => console.log("error", e));
+    }
+  };
+
+  private forgotPassword = async(request: Request, response: Response, next: NextFunction) =>{
+    const userData = request.body;
+    this.resettingPassword.resetPassword(userData);
+  }
+
+  private verifyUser = async (request: Request, response: Response, next: NextFunction)=>{
+    this.user.findOne({
+      confirmationCode: request.params.confirmationCode,
+    })
+      .then((user) => {
+        if (!user) {
+          return response.status(404).send({ message: "User Not found." });
+        }
+  
+        user.status = "Active";
+        user.save((err) => {
+          if (err) {
+            response.status(500).send({ message: err });
```

Comment:
500-tka oznacza błąd nie obsłużony na backendzie do czego nie możemy dopuścić. Co dokładnie miałaś tutaj na myśli/ chciałaś zrobić zwracając taki response? Czy chodziło o coś takiego? https://openclassrooms.com/en/courses/5614116-go-full-stack-with-node-js-express-and-mongodb/5656221-save-and-retrieve-data